Mathlib/Algebra/Category/ModuleCat/Presheaf.lean

```diff
@@ -497,4 +497,81 @@ def unitHomEquiv (M : PresheafOfModules R) :
     ext X
     exact (LinearMap.ringLmapEquivSelf (R.obj X) ℤ (M.obj X)).apply_symm_apply (s.val X)
 
+section module_over_initial
+
+/-!
+## `PresheafOfModules R ⥤ Cᵒᵖ ⥤ ModuleCat (R.obj X)` when `X` is initial
+
+When `X` is initial, we have `Module (R.obj X) (M.obj c)` for any `c : Cᵒᵖ`.
+
+-/
+
+/--
+Implementation of the functor `PresheafOfModules R ⥤ Cᵒᵖ ⥤ ModuleCat (R.obj X)`
+when `X` is initial.
+
+The functor is implemented as, on object level `M ↦ (c ↦ M(c))` where the `R(X)`-module structure
+on `M(c)` is given by restriction of scalars along the unique morphism `R(c) ⟶ R(X)`; and on
+morphism level `(f : M ⟶ N) ↦ (c ↦ f(c))`.
+-/
+@[simps]
+noncomputable def forgetToPresheafModuleCatObj
+    (X : Cᵒᵖ) (hX : Limits.IsInitial X) (M : PresheafOfModules.{v} R) :
+    Cᵒᵖ ⥤ ModuleCat (R.1.obj X) :=
+{ obj := fun c =>
+      letI : Module (R.1.obj X) (M.presheaf.obj c) :=
+        Module.compHom (M.presheaf.obj c) (R.1.map (hX.to c))
+      ModuleCat.of _ (M.presheaf.obj c)
```

Comment:
Here, you should use `ModuleCat.restrictScalars` applied to the obvious map of rings and the object `M.obj c`.

Reply:
Thanks!